src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/AceBackgroundHighlighter.java

```diff
@@ -445,7 +445,7 @@ private static List<HighlightPattern> sweaveHighlightPatterns()
    {
       return ListUtil.create(
             new HighlightPattern(
-                  "<<(.*?)>>",
+                  "^\\s*<<.*>>=\\s*$",
```

Comment:
Also note that Sweave chunks should have format `<<>>=`, and we were previously missing the trailing `=`.